File: app/Console/Commands/PayrollTick.php
<?php

namespace App\Console\Commands;

use App\Services\Payroll\PayrollRunner;
use Illuminate\Console\Command;

class PayrollTick extends Command
{
    protected $signature = 'payroll:tick';

    protected $description = 'Keep the current/next payroll periods ready and, on pay day, remind everyone who runs payroll.';

    public function handle(PayrollRunner $runner): int
    {
        $r = $runner->tick();
        $this->info("Periods created: {$r['periods_created']} · Reminded: ".($r['reminded'] ? implode(', ', $r['reminded']) : 'none'));

        return self::SUCCESS;
    }
}

File: payroll/management/commands/payroll_tick.py
from django.core.management.base import BaseCommand

from payroll.runner import PayrollRunner


class Command(BaseCommand):
    help = "Keep the current/next payroll periods ready and, on pay day, remind everyone who runs payroll."

    def handle(self, *args, **options):
        r = PayrollRunner().tick()
        self.stdout.write(f"Periods created: {r['periods_created']} · Reminded: {', '.join(r['reminded']) or 'none'}")
